File: solutions/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n=nums.length;
        int maxsum=Integer.MIN_VALUE;
        int csum=0;
        for(int i=0;i<n;i++){
            csum+=nums[i];
            if(maxsum<csum) maxsum =csum;
            if(csum<0){
                csum=0;
            }
        }
        return maxsum;
    }
}

File: solutions/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum=0
        maxi=-1000000000
        for i in nums:
            sum+=i
            maxi=max(maxi,sum)
            if sum<0:
                sum=0
        return maxi

        
